File: client_StockCheck_KotlinEngineers/app/src/main/java/com/cstp2205_s25/client_stockcheck_kotlinengineers/data/entities/InventoryItem.kt
package com.cstp2205_s25.client_stockcheck_kotlinengineers.data.entities

data class InventoryItem (
    val id: String? = null, // DB is going to generate
    val name: String = "",
    val description: String = "",
    val category: String = "",
    val qty: Int = 0,
    val price: Double = 0.0,
    val imageUrl: String = "",
    val locationId: String? = null,
    val status: String? = null,
)

File: server_StockCheck_KotlinEngineers/models/InventoryItem.js
const mongoose = require('mongoose');

const inventoryItemSchema = new mongoose.Schema({
  name: { type: String, required: true },
  description: { type: String, default: "" },
  category: { type: String, default: "" },
  qty: { type: Number, required: true },
  price: { type: Number, default: 0.0 },
  imageUrl: { type: String, default: "" },
  locationId: { type: mongoose.Schema.Types.ObjectId, ref: 'Location', required: false },
  status: { type: String, enum: ['In Stock', 'Out of Stock'], default: 'In Stock' }
}, { timestamps: true });

module.exports = mongoose.model('InventoryItem', inventoryItemSchema);
